includes/Modules/Analytics_4/Settings.php

```diff
@@ -95,23 +122,76 @@ protected function get_sanitize_callback() {
 				}
+
 				if ( isset( $option['googleTagID'] ) ) {
 					if ( ! preg_match( '/^(G|GT|AW)-[a-zA-Z0-9]+$/', $option['googleTagID'] ) ) {
 						$option['googleTagID'] = '';
 					}
 				}
-				if ( isset( $option['googleTagAccountID'] ) ) {
-					if ( ! is_numeric( $option['googleTagAccountID'] ) || ! $option['googleTagAccountID'] > 0 ) {
-						$option['googleTagAccountID'] = '';
-					}
-				}
-				if ( isset( $option['googleTagContainerID'] ) ) {
-					if ( ! is_numeric( $option['googleTagContainerID'] ) || ! $option['googleTagContainerID'] > 0 ) {
-						$option['googleTagContainerID'] = '';
+
+				$numeric_properties = array( 'googleTagAccountID', 'googleTagContainerID' );
+				foreach ( $numeric_properties as $numeric_property ) {
+					if ( isset( $option[ $numeric_property ] ) ) {
+						if ( ! is_numeric( $option[ $numeric_property ] ) || ! $option[ $numeric_property ] > 0 ) {
+							$option[ $numeric_property ] = '';
+						}
 					}
 				}
 			}
+
 			return $option;
 		};
 	}
+
+	/**
+	 * Updates settings to set the current user as an owner of the module if settings contain owned keys.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings The module settings.
+	 */
+	protected function maybe_set_owner_id( $settings ) {
+		$keys = $this->get_owned_keys();
+		if ( count( array_intersect( array_keys( $settings ), $keys ) ) > 0 && current_user_can( Permissions::MANAGE_OPTIONS ) ) {
+			$user_id = get_current_user_id();
+			$this->merge( array( 'ownerID' => $user_id ) );
+
+			// This is a temporary solution to sync owner IDs between Analytics and Analytics 4 modules.
+			// The owner ID setting of the Analytics module is the source of truth for the Analytics 4 module.
+			// This will change when Analytics is sunset and we merge both modules into one.
+			( new Analytics_Settings( $this->options ) )->merge( array( 'ownerID' => $user_id ) );
+		}
+	}
+
+	/**
+	 * Updates the current module settings to have the current user as the owner of the module if at least
+	 * one of the owned keys have been changed.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings The new module settings.
+	 * @param array $old_settings The old module settings.
+	 */
+	protected function maybe_update_owner_id_in_settings_before_saving_them( $settings, $old_settings ) {
+		$keys = $this->get_owned_keys();
+		foreach ( $keys as $key ) {
+			if (
+				isset( $settings[ $key ], $old_settings[ $key ] ) &&
+				$settings[ $key ] !== $old_settings[ $key ] &&
+				current_user_can( Permissions::MANAGE_OPTIONS )
+			) {
+				$settings['ownerID'] = get_current_user_id();
+
+				// This is a temporary solution to sync owner IDs between Analytics and Analytics 4 modules.
+				// The owner ID setting of the Analytics module is the source of truth for the Analytics 4 module.
+				// This will change when Analytics is sunset and we merge both modules into one.
+				( new Analytics_Settings( $this->options ) )->merge( array( 'ownerID' => $settings['ownerID'] ) );
```

Comment:
I have to say, I'm not such a fan of doing a DB write within a `pre_update_option_` filter hook, where there's no guarantee the `Analytics_4` setting will actually save, but I guess it's unlikely the `merge()` would succeed while the subsequent `Analytics_4` update fails...